packages/fxa-auth-server/fxa-oauth-server/lib/keys.js

```diff
@@ -15,37 +141,58 @@ function pub(key) {
   // BE CAREFUL IF YOU REFACTOR THIS. Thanks.
   return {
     kty: key.kty,
-    alg: 'RS256',
+    alg: key.alg || 'RS256',
     kid: key.kid,
     'fxa-createdAt': key['fxa-createdAt'],
-    use: 'sig',
+    use: key.use || 'sig',
     n: key.n,
     e: key.e,
   };
-}
-
-const PRIVATE_JWKS_MAP = new Map();
-
-const currentPrivJWK = config.get('openid.key');
-PRIVATE_JWKS_MAP.set(currentPrivJWK.kid, currentPrivJWK);
-
-const oldPrivJWK = config.get('openid.oldKey');
-if (oldPrivJWK && Object.keys(oldPrivJWK).length) {
-  PRIVATE_JWKS_MAP.set(oldPrivJWK.kid, oldPrivJWK);
-}
-
-const SIGNING_JWK = config.get('openid.key');
-const SIGNING_PEM = jwk2pem(SIGNING_JWK);
-
-const PUBLIC_JWK_MAP = new Map();
-const PUBLIC_PEM_MAP = new Map();
-
-PRIVATE_JWKS_MAP.forEach((privJWK, kid) => {
-  const publicJWK = pub(privJWK);
+};
 
-  PUBLIC_JWK_MAP.set(kid, publicJWK);
-  PUBLIC_PEM_MAP.set(kid, jwk2pem(publicJWK));
-});
+/**
+ * Generate a new, random private key.
+ *
+ * @returns {JWK}
+ */
+exports.generatePrivateKey = function generatePrivateKey() {
+  const PEM_ENCODING = {
+    type: 'pkcs1',
+    format: 'pem',
+  };
+  const kp = crypto.generateKeyPairSync('rsa', {
+    modulusLength: 256 * 8,
+    publicKeyEncoding: PEM_ENCODING,
+    privateKeyEncoding: PEM_ENCODING,
+  });
+  // We tag our keys with their creation time, and a unique key id
+  // based on a hash of the public key and the timestamp.  The result
+  // comes out like:
+  //  {
+  //    kid: "20170316-ebe69008"
```

Comment:
@shane-tomlinson I went ahead and shorted the kids here in service of #1797, thoughts?

Reply:
Looks perfect to me!